src/app/visit/visit.cpp

```diff
@@ -0,0 +1,59 @@
+#include "visit.hpp"
+#include "../doctor/doctor.hpp"
+
+std::set<Visit *> Visit::visitExtent;
+
+Visit::Visit()
+{
+    visitExtent.insert(this);
+}
+
+Visit::~Visit()
+{
+    removeFromExtent(this);
+}
+
+std::shared_ptr<Visit> Visit::createVisit(std::shared_ptr<Doctor> doc)
+{
+    auto visit = std::shared_ptr<Visit>(new Visit());
```

Comment:
Please use std::make_shared here if possible.

Reply:
No, it must be implemented this way to avoid circular reference errors.

Reply:
To be honest, I can't see how this is connected to possible circular reference errors. 
I think, that shared_ptr is a shared_ptr regardles od the fact, how it is created (with new passed to constructor or std::make_shared).
Of course, there are subtle differences, but one way over other does not protect us more as far as I know.

I think, that for protection against circular dependencies is mainly done via weak_ptr's.
I